packages/react-sdk/src/component-generator.test.ts

```diff
@@ -548,3 +620,64 @@ test("generate page component with variables and actions", () => {
     `)
   );
 });
+
+test("avoid generating collection parameter variable as state", () => {
+  expect(
+    generatePageComponent({
+      scope: createScope(),
+      rootInstanceId: "body",
+      instances: new Map([
+        createInstancePair("body", "Body", [{ type: "id", value: "list" }]),
+        createInstancePair("list", collectionComponent, []),
+      ]),
+      dataSources: new Map([
+        createDataSourcePair({
+          id: "dataSourceList",
+          scopeInstanceId: "list",
+          type: "variable",
+          name: "data",
+          value: { type: "json", value: ["apple", "orange", "mango"] },
+        }),
+        createDataSourcePair({
+          id: "dataSourceItem",
+          scopeInstanceId: "list",
+          type: "variable",
+          name: "element",
+          value: { type: "json", value: `` },
+        }),
+      ]),
+      props: new Map([
+        createPropPair({
+          id: "propData",
+          instanceId: "list",
+          name: "data",
+          type: "expression",
+          value: "$ws$dataSource$dataSourceList",
+        }),
+        createPropPair({
+          id: "propItem",
+          instanceId: "list",
+          name: "item",
+          type: "parameter",
+          value: "dataSourceItem",
+        }),
+      ]),
+      indexesWithinAncestors: new Map(),
+    })
+  ).toEqual(
+    clear(`
```

Comment:
why not to use just inline snapshot?

Reply:
Until prettier v3 is supported very annoying to fix formatting manually all the time.